testing/integration/otel_test.go

```diff
@@ -0,0 +1,306 @@
+        regex: '^(?P<time>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}) (?P<sev>[A-Z]*) (?P<msg>.*)$'
+        timestamp:
+          parse_from: attributes.time
+          layout: '%%Y-%%m-%%d %%H:%%M:%%S'
+        severity:
+          parse_from: attributes.sev
+
+processors:
+  resource:
+    attributes:
+    # APM Server will use service.name for data stream name: logs-apm.app.<service_name>-default
+    - key: service.name
+      action: insert
+      value: elastic-otel-test
+    - key: host.test-id
+      action: insert
+      value: %s
+
+exporters:
+  debug:
+    verbosity: detailed
+    sampling_initial: 10000
+    sampling_thereafter: 10000
+  otlp/elastic:
+      endpoint: "127.0.0.1:8200"
+      tls:
+        insecure: true
+
+service:
+  pipelines:
+    logs:
+      receivers: [filelog]
+      processors: [resource]
+      exporters:
+        - debug
+        - otlp/elastic`
+
+func TestFileProcessing(t *testing.T) {
+	define.Require(t, define.Requirements{
+		Group: Default,
+		Local: true,
+		OS: []define.OS{
+			// input path missing on windows
+			{Type: define.Linux},
+			{Type: define.Darwin},
+		},
+	})
+
+	t.Cleanup(func() {
+		_ = os.Remove(fileProcessingFilename)
+	})
+
+	fixture, err := define.NewFixture(t, define.Version())
+	require.NoError(t, err)
+
+	ctx, cancel := testcontext.WithDeadline(t, context.Background(), time.Now().Add(10*time.Minute))
+	defer cancel()
+	err = fixture.Prepare(ctx, fakeComponent, fakeShipper)
+	require.NoError(t, err)
+
+	// replace default elastic-agent.yml with otel config
+	// otel mode should be detected automatically
+	err = fixture.Configure(ctx, fileProcessingConfig)
+	require.NoError(t, err)
+
+	var fixtureWg sync.WaitGroup
+	fixtureWg.Add(1)
+	go func() {
+		defer fixtureWg.Done()
+		err = fixture.RunWithClient(ctx, false)
+	}()
+
+	require.Eventually(t,
+		func() bool {
+			// verify file exists
+			content, err := os.ReadFile(fileProcessingFilename)
+			return err == nil && len(content) > 0
+		},
+		5*time.Minute, 500*time.Millisecond,
+		"there should be exported logs by now")
+
+	cancel()
+	fixtureWg.Wait()
+	require.True(t, err == nil || err == context.Canceled || err == context.DeadlineExceeded, "Retrieved unexpected error: %s", err.Error())
+}
+
+func TestAPMIngestion(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: Default,
+		Stack: &define.Stack{},
+		Local: true,
+		OS: []define.OS{
+			// apm server not supported on darwin
+			{Type: define.Linux},
+		},
+	})
+
+	stackVersion := info.KibanaClient.GetVersion()
+	agentVersion, err := version.New(aVersion.Agent)
+	require.NoError(t, err, "failed to get agent version")
+
+	if stackVersion.LessThan(agentVersion) {
+		// at the transition period when new version of stack is not yet released
+		// and we use override with previous version this test would fail with
+		// events being produced by newer APM than stack version
+		t.Skip("agent version needs to be equal to stack version")
+	}
```

Comment:
@cmacknz @blakerouse I thought we ran into this sort of issue with other integration tests and solved it in a more global fashion, that is, in a way that would work for all integration tests and every test wouldn't need skip logic like what's implemented here?